flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultValidator.java

```diff
@@ -270,4 +271,37 @@
 
         return Optional.empty();
     }
+
+    // validate session job
+
+    @Override
+    public Optional<String> validate(
+            FlinkSessionJob sessionJob, Optional<FlinkDeployment> session) {
+
+        return firstPresent(
+                validateNotApplicationCluster(session),
+                validateSessionClusterId(sessionJob, session));
+    }
+
+    private Optional<String> validateSessionClusterId(
+            FlinkSessionJob sessionJob, Optional<FlinkDeployment> session) {
+        return session.map(
+                s -> {
+                    if (!s.getMetadata().getName().equals(sessionJob.getSpec().getClusterId())) {
+                        return "The session job's cluster id is not match with the session cluster";
+                    }
+                    return null;
+                });
+    }
+
+    private Optional<String> validateNotApplicationCluster(Optional<FlinkDeployment> session) {
+        return session.map(
+                s -> {
+                    if (s.getSpec().getJob() != null) {
+                        return "Can not submit to application cluster";
+                    } else {
+                        return null;
```

Comment:
Wouldn't this throw a nullpointer exception in the Optional logic?
I think this should be a `flatMap` and return Optional.of(error) or empty

Reply:
The null value can be handled by `Optional.ofNullable` in `map` function, but the `flatMap` way looks more clear, Adjusted accordingly.